main.py

```diff
@@ -83,39 +122,52 @@ def distro_weights(videos, requests):
     # print(ratings)
     return ratings
 
-def distribute_videos(videos, requests, endpoints):
+
+def distribute_videos(videos, endpoints):
     caches = [[] for _ in range(C)]
     i = 0
     for e in endpoints:
-        if e[2]:  # can find closest cache
-            closest_cache = e[2][0]
-            for cache in e[2]:
-                # find cache closest to e
-                if cache[1] < closest_cache[1]:
-                    # found cache closer
-                    closest_cache = cache
-
-            v_index = 0
-            # print(i, closest_cache)
-            for v in distro_weights(videos, requests):
-                # if video is most beneficial to this endpoint
-                if v[i] == max(v) and v[i] > 0:
-                    caches[closest_cache[0]] += [v_index]
-                v_index += 1
+        v_index = 0
+        print(i, '/', E)
+        # print(i, closest_cache)
+        for v in rated_videos:
+            # if video is most beneficial to this endpoint
+            if v[i] == max(v) and v[i] > 0:
+                # best_cache will be closest available, considering sizes of ones which
+                # are already hosting videos
+                best_cache = find_closest_available_cache(caches, e, videos[v_index])
+                # if best_cache exists
+                if best_cache:
+                    caches[best_cache] += [v_index]
+            v_index += 1
         else:  # no caches
-            pass
+                pass
+        
         i += 1
     return caches
 
-caches = distribute_videos(videos, requests, endpoints)
+
+def fill_space(videos, requests, endpoints):
+    pass
+
 
 #FORMAT TO VALID
+#example caches array with 5 caches and the videos they store, some store just one video, some none and some more
+# caches = ((1,0),(2,3,4),(5),(),(5,4,6,3))
+
+if not read_ratings_from_file:
+    r = open(rating_file, "w")
+    rated_videos = rate_videos_bulk(videos, requests)
+    r.write(str(rated_videos))
+    r.close()
```

Comment:
because rating videos always returns the same output, store that in a local file to test out the distribution algorithm immediately